writing-your-first-es2015-javascript-module.md

```diff
@@ -0,0 +1,140 @@
+  add this directory to source control).
+* `test/`: the test directory (the tests exercise the ES2015 code in the
+  `/src` directory rather than the transpiled code).
+
+## Unit Tests and Coverage
+
+Babel provides the module `babel-register` which hooks into Node's require statement. This hook compiles your ES2015 code on the fly as your project needs it. `babel-register` is great for writing unit-tests, allowing you to exercise your ES2015 code directly without executing your build step.
+
+The test coverage tool [nyc](https://github.com/bcoe/nyc) integrates with `babel-register`, allowing you to add coverage reporting and transpilation in one fell swoop. Here's how this can be achieved:
+
+1. install your favorite unit testing framework (in my case `npm i mocha --save-dev`).
+2. install `nyc` (`npm i nyc --save-dev`), `nyc` will be used both for instrumenting
+  test coverage, and for compiling ES2015 code on the fly.
+3. add a `test` script to your `package.json` that looks something like this:
+
+```sh
+nyc --require=babel-core/register mocha
+```
+
+The flag `--require babel-core/register` tells nyc to include `babel-register` when executing the
+test suite, eliminating the need for a transpilation step when running your tests.
+
+## Packaging Your ES2015 Module
+
+It's nice to be able to run your unit-tests against the raw ES2015 code, but when publishing
+a module you want it to work on a variety of platforms, we achieve this by adding a `prepublish`
+build step to our ES2015 module. I'm a big fan of using npm scripts for this sort of thing.
+
+Here are the important parts of `yarsay`'s _package.json_:
+
+```json
+{
+  "scripts": {
+    "build": "cd src; babel ./*.js -d ../lib",
+    "prepublish": "npm run build",
+    "pretest": "standard ./src/*.js",
+    "test": "nyc --require babel-core/register mocha"
+  },
+  "main": "lib/yarsay.js",
+  "bin": "lib/cli.js"
+}
+```
+
+* `build`: enters the ES2015 source directory and executes the transpiler, and stores the final source-code in `/lib`.
+* `prepublish`: executes when you publish a new version of the module and simply runs the build step.
+* `test`: uses nyc to run the test suite, compiling ES2015 code on the fly.
+* `pretest`: I'm a fan of the [standard](https://github.com/feross/standard) code-style tool, and generally run this as a `pretest` hook.
+* `main`/`bin`: note that `main` and `bin` both reference the compiled code in the `lib/` folder.
+
+When we run `npm publish`, the build script executes, and we ultimately publish the cross-platform ES5 source code.
+
+## Now Go Forth
+
+That's all there is to it! using a few npm scripts, `nyc`, and our favorite test runner, we've created
+an environment that allows us to write and test code in ES2015, while publishing a modules that work on
+legacy versions of Node.js.
+
+I encourage you to play with [yarsay](https://github.com/bcoe/yarsay) and use it as a jumping off point
```

Comment:
[clarify] How about "use it as a launchpad" instead of "use it as a jumping off point"?